scripts/dts/gen_defines.py

```diff
@@ -599,6 +603,44 @@ def write_clocks(node):
                 controller.props["clock-frequency"].val)
 
 
+def write_dmas(node):
+    if "dmas" not in node.props:
+        return
+
+    write_phandle_val_list(node.props["dmas"])
+
+    # Client parameter syntax may vary with the DMA controller changes.
+    for dma_i, dma in enumerate(node.props["dmas"].val):
+        if "st,stm32-dma" in dma.controller.compats:
```

Comment:
Question to @ulfalizer for my understanding: I'm actually surprised we need to modify `gen_defines.py` script for compatible specific treatment. I would have expected yaml bindings help us to generate whatever is needed in a genric way.
Having defined `dma_cells` in bindings isn't it sufficient to generate matching defines?
How does it differ with gpio or clock cells?

Reply:
Seems it's checking if particular bits in values in `(values)` in `dmas = <&ctrl (values)>` are set to determine which #defines to generate. No way to specify that in a binding. Would need to add a complete programming language to bindings to handle all the ways #defines might be generated.

Don't know anything about this controller, so don't know if it ought to be handled differently.

Reply:
Another option would be to move the bit extraction stuff to the C code. Then the generic `phandle-array` handling could be used here.

Reply:
The main reason why this is different from gpio or clock cells is that we want its dts parameter format to be consistent to the stm32 DMA consumer parameters format declared in Linux. If not, it would easy to declare more cells for each parameter and remove these decoding code.

Reply:
Could have stuff like

    #define SOME_FLAG (DT_BLAH & 4)

in the fixup file too (or directly in the `.c` file... hope we can get rid of fixup files some day). That's still testable at compile time.

I could go either way here at least.

Reply:
> Another option would be to move the bit extraction stuff to the C code. Then the generic phandle-array handling could be used here.

I prefer this solution as well. 
I understand having this kind of "compatible" specific code could be handy, but it breaks the whole purpose of generic dts extraction script.